blocks/log.py

```diff
@@ -64,10 +69,3 @@ def previous_row(self):
     @property
     def last_epoch_row(self):
         return self[self.status['_epoch_ends'][-1]]
-
-    def to_dataframe(self):
-        """Convert a log into a :class:`.DataFrame`."""
-        if not PANDAS_AVAILABLE:
-            raise ImportError("The pandas library is not found. You can"
-                              " install it with pip.")
-        return DataFrame.from_dict(self, orient='index')
```

Comment:
I don't agree with these arguments. Two types of dependencies should be distinguished, those on which significant amount of code depends, and those, with which Blocks supports only limited cooperation, export being an example. I think there can be as many as required dependencies of the second type, provided that they are not mandatory (i.e. not installed by default and not necessary most of the time) and that when they are missing, a meaningful error with an installation instruction is raised.

I also would say that using pandas makes much more sense than oneliners like yours, and my educated guess is that many from the Blocks audience use it. We can of course take a survey :)

Finally `SQLite` backend is coming, and having `to_dataframe` becomes more important. 

Also I think that this PR was merged too quickly, and that further PRs that remove functionality from Blocks are held a bit longer. It is reasonable to give the community at least three days to protest, e.g. I  simply didn't have time to reply. It took you and David less the 24 hours to merge it. I propose to return `to_dataframe` and `pandas` as optional dependency.